lms/djangoapps/grades/rest_api/v1/gradebook_views.py

```diff
@@ -506,7 +508,19 @@ def _gradebook_entry(self, user, course, graded_subsections, course_grade):
 
     @staticmethod
     def _get_external_user_key(user, course_id):
-        program_enrollment = CourseEnrollment.get_program_enrollment(user, course_id)
+        program_course_enrollments = ProgramCourseEnrollment.objects.filter(
+            course_enrollment__user=user,
+            course_key=course_id
+        )
+
+        if not program_course_enrollments:
+            return None
+
+        program_enrollment = program_course_enrollments.first().program_enrollment
+        for program_course_enrollment in program_course_enrollments:
+            if program_course_enrollment.status == ProgramCourseEnrollmentStatuses.ACTIVE:
+                program_enrollment = program_course_enrollment.program_enrollment
```

Comment:
The logic here is a little strange:

*Select the __last__ active program course enrollment in the queryset, unless there are no active ones, in which case select the __first__ program course enrollment in the queryset.*

Is how we want to handle the edge cases of there being zero or multiple active program course enrollments? If so, please explain that in a comment, and if not, please make the edge-case-handling logic more consistent and log/raise where necessary.

I also recommend calling this `get_external_student_external_key_by_course` and making it part of the `program_enrollments` public reading API, because:
* Program enrollments business logic should stay within `program_enrollments` app.
* Other apps may also want to a user's get external key by course, and it'd be a shame if they duplicated that logic or imported from the gradebook API.